server/src/main.rs

```diff
@@ -85,18 +90,36 @@ impl LanguageServer for Backend {
     }
 
     async fn initialized(&self, _: InitializedParams) {
-        let workspace_uri = self.workspace_uri.lock().await;
+        // Update the sources and references directories
+        {
+            let workspace_uri = self.workspace_uri.lock().await;
+
+            // Fetch and set the sources directory
+            let sources_directory = self.get_sources_directory(&workspace_uri, None).await.ok();
+            let references_directories = self
+                .get_reference_directories(&workspace_uri, None)
+                .await
+                .ok();
+            let mut sources_uri = self.sources_uri.lock().await;
+            let mut references_uris = self.reference_uris.lock().await;
+            *sources_uri = sources_directory;
+            *references_uris = references_directories;
+        }
+
+        let mut shared_state_lock = self.shared_state.lock().await;
+
+        // Compile the Slice files and publish diagnostics
+        let (updated_state, options) = self.compile_slice_files().await;
+
+        shared_state_lock.compilation_state = updated_state;
+        shared_state_lock.compilation_options = options;
+
+        self.publish_diagnostics_for_all_files(&mut shared_state_lock)
+            .await;
 
-        // Fetch and set the sources directory
-        let sources_directory = self.get_sources_directory(&workspace_uri, None).await.ok();
-        let references_directories = self
-            .get_reference_directories(&workspace_uri, None)
-            .await
-            .ok();
-        let mut sources_uri = self.sources_uri.lock().await;
-        let mut references_uris = self.reference_uris.lock().await;
-        *sources_uri = sources_directory;
-        *references_uris = references_directories;
+        self.client
+            .log_message(MessageType::INFO, "Slice language server initialized")
```

Comment:
We have a mix of `Slice language server` and `Slice Language Server`. We should be consistent. 

```
❯ rg -i 'Slice language server'
client/package.json
3:  "description": "VSCode client for the Slice language server",

client/src/extension.ts
14:  "Slice Language Server trace"
32:    "Slice Language Server",
130:      "Slice Language Server failed to start. See the trace for more information."

README.md
9:Additionally, this extension provides a Slice language server, for `.slice` files only.

server/src/main.rs
108:            .log_message(MessageType::INFO, "Slice language server config changed")
143:                    "Failed to update sources directory. Please check your Slice language server configuration.",
```